### PR TITLE
[RW-7734][risk=low] Bump gatk jar override version

### PR DESCRIPTION
**Pull request by calbach**
Cherry-picked log4j upgrade onto the prior gatk jar version.

Currently testing this locally.

### CODE REVIEW

**Review by yonghaoy — APPROVED** [review]

File: api/config/config_preprod.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "extractionCohortsDataset": "fc-aou-cdr-preprod-ct.wgs_extraction_cohorts",
     "extractionDestinationDataset": "fc-aou-cdr-preprod-ct.wgs_extraction_destination",
     "extractionTempTablesDataset": "fc-aou-cdr-preprod-ct.wgs_extraction_temp_tables",
+    "gatkJarUri": "gs:\/\/all-of-us-workbench-test-genomics\/wgs\/gatk-package-4.2.0.0-131-g2c1fc16-SNAPSHOT-local.jar",
```
Comment (calbach) [inline api/config/config_preprod.json:58]: FYI: this dependency is not ideal, but it is the existing behavior. https://precisionmedicineinitiative.atlassian.net/browse/RW-7261 tracks a fix there (I need to create a new bucket and configure permissions to resolve this - which is why I'm not changing it here)

Comment (yonghaoy) [reply api/config/config_preprod.json:58]: I don't have enough context, but another direction that we might be able to explore is to use the jar hosted in Broad jfrog. Add a gradle task that download jar from jfrog, and put is under `resource/`

Example: (it did a step further, i..e unzip that jar file)
https://github.com/DataBiosphere/terra-resource-buffer/blob/master/build.gradle#L193L204

Comment (calbach) [reply api/config/config_preprod.json:58]: Thanks. The GATK jar is needed in a Cromwell workflow - not on the spring server, though pushing the jar to a package repo makes sense. Trying to figure this out with the Broad team.

Comment (yonghaoy) [reply api/config/config_preprod.json:58]: Thanks. I did see those jars in Broad's jfrog, e.g. : https://broadinstitute.jfrog.io/artifactory/libs-snapshot-local/org/broadinstitute/gatk/4.2.0.0-36-gc84eaee-SNAPSHOT/gatk-4.2.0.0-36-gc84eaee-20210609.142057-1.jar

**Comment by calbach** [comment]
This is built off a branch on the gatk repo, it's not the main release: https://github.com/broadinstitute/gatk/tree/ah_var_store

**Comment by calbach** [comment]
Still trying to validate this locally. I'm now just trying to rebuild at HEAD rather than cherrypick. We did not do a good job of documenting the build process last time this happened.

**Comment by calbach** [comment]
Yes! this finally ran successfully on this version. Updated the ticket with the details.

**Pull request merged**